radix-transactions/src/model/ledger_transaction.rs

```diff
@@ -21,7 +21,46 @@ const ROUND_UPDATE_V1_LEDGER_TRANSACTION_DISCRIMINATOR: u8 = 2;
 const FLASH_V1_LEDGER_TRANSACTION_DISCRIMINATOR: u8 = 3;
 const USER_V2_LEDGER_TRANSACTION_DISCRIMINATOR: u8 = 4;
 
-define_raw_transaction_payload!(RawLedgerTransaction);
+enum LedgerTransactionKind {
+    Genesis,
+    User,
+    Validator,
+    ProtocolUpdate,
+}
+
+impl LedgerTransactionKind {
+    fn discriminator_for_hash(&self) -> u8 {
+        match self {
+            LedgerTransactionKind::Genesis => GENESIS_LEDGER_TRANSACTION_DISCRIMINATOR,
+            LedgerTransactionKind::User => USER_V1_LEDGER_TRANSACTION_DISCRIMINATOR,
```

Comment:
Looks like we're reusing the same discriminator for v2? is this intended? If so, should we use a separate, version-agnostic const for that? Or perhaps add a comment.

Reply:
Yeah, this is for backwards compatibility. Will add a comment :).